examples/demo2/tabs/traceroute.rs

```diff
@@ -0,0 +1,194 @@
+use ratatui::{
+    prelude::*,
+    widgets::{
+        canvas::{Canvas, Map, Points},
+        *,
+    },
+};
+
+use super::Tab;
+use crate::{colors, styles, tui::layout};
+
+#[derive(Debug)]
+struct Hop {
+    host: &'static str,
+    address: &'static str,
+    location: (f64, f64),
+}
+
+impl Hop {
+    const fn new(name: &'static str, address: &'static str, location: (f64, f64)) -> Self {
+        Self {
+            host: name,
+            address,
+            location,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct TracerouteTab {
+    selected_row: usize,
+}
+
+impl TracerouteTab {
+    pub fn new(selected_row: usize) -> Self {
+        Self {
+            selected_row: selected_row % HOPS.len(),
+        }
+    }
+}
+
+impl Tab for TracerouteTab {
+    fn title(&self) -> String {
+        "Traceroute".to_string()
+    }
+
+    fn render(&self, area: Rect, buf: &mut Buffer) {
+        self.render_traceroute_tab(area, buf);
+    }
+
+    fn select(&mut self, row: usize) {
+        self.selected_row = row;
+    }
+}
+
+impl TracerouteTab {
+    fn render_traceroute_tab(&self, area: Rect, buf: &mut Buffer) {
+        colors::render_rgb_colors(area, buf);
+        let area = area.inner(&Margin {
+            vertical: 1,
+            horizontal: 2,
+        });
+        Clear.render(area, buf);
+        Block::new().style(styles::APP).render(area, buf);
+        let area = Layout::default()
+            .direction(Direction::Horizontal)
+            .constraints(vec![Constraint::Ratio(1, 2), Constraint::Ratio(1, 2)])
+            .split(area);
+        let left_area = layout(area[0], Direction::Vertical, vec![0, 4]);
+        self.render_hops(left_area[0], buf);
+        render_ping(self.selected_row, left_area[1], buf);
+        self.render_map(area[1], buf);
+    }
+
+    fn render_hops(&self, area: Rect, buf: &mut Buffer) {
+        let mut state = TableState::default().with_selected(Some(self.selected_row));
+        let rows = HOPS
+            .iter()
+            .map(|hop| Row::new(vec![hop.host, hop.address]))
+            .collect_vec();
+        let block = Block::default()
+            .title("Traceroute bad.horse")
+            .borders(Borders::ALL)
+            .border_type(BorderType::Rounded);
+        StatefulWidget::render(
+            Table::new(rows)
+                .header(Row::new(vec!["Host", "Address"]).bold().underlined())
+                .widths(&[Constraint::Max(100), Constraint::Length(15)])
+                .highlight_style(Style::new().dark_gray().on_white())
+                .block(block),
+            area,
+            buf,
+            &mut state,
+        );
+    }
+
+    fn render_map(&self, area: Rect, buf: &mut Buffer) {
+        let path: Option<(&Hop, &Hop)> = HOPS.iter().tuple_windows().nth(self.selected_row);
+        let block = Block::new().title("Map").borders(Borders::ALL);
+        let map = Map {
+            resolution: canvas::MapResolution::High,
+            color: Color::Gray,
```

Comment:
I think it'll look more consistent if the email tab's email content section matched this color.